### PR TITLE
Add copyResources option to suppress resource copying

### PR DESCRIPTION
**Pull request by yufeih**
For docs publish, we don't need to copy the resources. This PR:

- renames _asset_ to _resource_
- add _copyResources_ config
- add _sourcePath_ to build.manifest

#2972

### CODE REVIEW

File: docs/designs/output.md [hunk 1]
```diff
@@ -127,14 +127,19 @@ These are called *browser navigation properties* and are stored in manifest.
...
         "outputPath": "en-us/netstandard-2.0/dotnet/api/system.string.json",
         "locale": "en-us",
         "moniker": "netstandard-2.0", 
 
         // other browser navigation properties...
         // other properties needed for backward compatibility
+    }, {
+        "siteUrl": "/dotnet/api/system.string",
+
+        // when `outputPath` does not exists,
```
Comment (superyyrrzz) [inline docs/designs/output.md]: >exists [](start = 38, length = 6)

exist

File: src/docfx/build/Build.cs [hunk 2]
```diff
@@ -63,7 +63,13 @@ internal static class Build
 
                 bool ShouldBuildFile(Document file)
                 {
-                    return file.ContentType != ContentType.Unknown && fileListBuilder.TryAdd(file);
+                    if (file.ContentType == ContentType.Unknown)
+                        return false;
+                    if (!fileListBuilder.TryAdd(file))
+                        return false;
+                    if (file.ContentType == ContentType.Resource && !file.Docset.Config.Output.CopyResources)
+                        return false;
```
Comment (superyyrrzz) [inline src/docfx/build/Build.cs]: Will we build metadata for resource? If so, resource *should* be built when copyResources=true.

Comment (OsmondJiang) [inline src/docfx/build/Build.cs]: Move this logic to BuildResource?

File: docs/designs/output.md [hunk 3]
```diff
@@ -127,14 +127,19 @@ These are called *browser navigation properties* and are stored in manifest.
...
     "files": [{
-        "source": "dotnet/api/system.string",
         "siteUrl": "/dotnet/api/system.string",
         "outputPath": "en-us/netstandard-2.0/dotnet/api/system.string.json",
         "locale": "en-us",
         "moniker": "netstandard-2.0", 
 
         // other browser navigation properties...
         // other properties needed for backward compatibility
+    }, {
```
Comment (OsmondJiang) [inline docs/designs/output.md]: >}, { [](start = 4, length = 4)

format?

File: src/docfx/build/Build.cs (hunk 2 above)
Comment (yufeih) [reply src/docfx/build/Build.cs]: yeah, the only downside is progress, it will immediately jump from 30% to 100%

Comment (yufeih) [reply src/docfx/build/Build.cs]: we don't build resource metadata, it is just a copy, but it is a good point though, putting it here will cause other problems.

**Pull request merged**